lab-abigail/controller/recipe-control.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const Promise = require('bluebird');
+const createError = require('http-errors');
+const Recipe = require('../model/recipe-model');
+
+module.exports = exports = {};
+
+exports.createItem = function(req, res, recipe) {
+
+  if(!recipe) return Promise.reject(createError(400, 'bad requst'));
+
+  new Recipe(req.body).save()
+  .then(recipe => {
+    res.json(recipe);
```

Comment:
**TODO FOR LOST POINTS**

Your POST test failed due to a timeout error and it's because you didn't add an explicit ``return`` at Line 15. When doing multi-line ``then/catch`` blocks, you need to add the word ``return`` or else it'll just hang and not actually return anything. One-liners don't need to have ``return``. I.e.:

```
This is OK
.then(recipe => res.json(recipe))

But breaking this out means you put the word "return"

.then(recipe => {
   return res.json(recipe);
})
```